tests/integration/helpers.py

```diff
@@ -63,6 +64,32 @@ def _install_proxy(conn: SSHConnection, proxy: ProxyConfig | None = None):
     assert result.ok, "Failed to configure iptable rules"
 
 
+def _configure_dockerhub_mirror(conn: SSHConnection, dockerhub_mirror: str | None):
```

Comment:
Why is a host config change implemented in a helper instead of being in the contributing guide and in the github workflow ? This seems wrong to me

Reply:
If I understood the question correctly, the question is why this not setup during github workflow - the reason is because we're spawning bare Openstack machines with the image that was created by the charm. One of the limitations of the current testing approach is that the dockerhub mirror is not configured on the image side (the configuration value is set by the GitHub runner charm configuration and applied dynamically on the runners). This made the integration tests hit dockerhub rate limit issues. By adding this (the same approach GitHub runner charm sets up dockerhub mirror!), it allows using dockerhub mirror for tests :D

Reply:
Got it